language-support/hs/bindings/src/DA/Ledger/Services/TransactionService.hs

```diff
@@ -17,7 +25,105 @@ import Network.GRPC.HighLevel.Generated
 import qualified Com.Digitalasset.Ledger.Api.V1.TransactionService as LL
 import qualified Data.Map as Map
 
--- TODO:: all the other RPCs
+getTransactions :: GetTransactionsRequest -> LedgerService (Stream Transaction)
+getTransactions req =
+    makeLedgerService $ \timeout config -> do
+    stream <- newStream
+    _ <- forkIO $
+        withGRPCClient config $ \client -> do
+            service <- LL.transactionServiceClient client
+            let LL.TransactionService {transactionServiceGetTransactions=rpc} = service
+            sendToStreamFlat timeout (lowerRequest req) f stream rpc
+    return stream
+    where f = raiseList raiseTransaction . LL.getTransactionsResponseTransactions
+
+getTransactionTrees :: GetTransactionsRequest -> LedgerService (Stream TransactionTree)
+getTransactionTrees req =
+    makeLedgerService $ \timeout config -> do
+    stream <- newStream
+    _ <- forkIO $
+        withGRPCClient config $ \client -> do
+            service <- LL.transactionServiceClient client
+            let LL.TransactionService {transactionServiceGetTransactionTrees=rpc} = service
+            sendToStreamFlat timeout (lowerRequest req) f stream rpc
+    return stream
+    where f = raiseList raiseTransactionTree . LL.getTransactionTreesResponseTransactions
+
+getTransactionByEventId :: LedgerId -> EventId -> [Party] -> LedgerService (Maybe TransactionTree)
+getTransactionByEventId lid eid parties =
+    makeLedgerService $ \timeout config -> do
+    withGRPCClient config $ \client -> do
+        service <- LL.transactionServiceClient client
+        let LL.TransactionService{transactionServiceGetTransactionByEventId=rpc} = service
+        rpc (ClientNormalRequest (mkByEventIdRequest lid eid parties) timeout emptyMdm)
+        >>= \case
+            ClientNormalResponse (LL.GetTransactionResponse Nothing) _m1 _m2 _status _details ->
+                fail "GetTransactionResponse, transaction field is missing"
+            ClientNormalResponse (LL.GetTransactionResponse (Just tx)) _m1 _m2 _status _details ->
+                case raiseTransactionTree tx of
+                    Left reason -> fail (show reason)
```

Comment:
Maybe also add in the error message that it originated from `raiseTransactionTree`.

Reply:
What I really want to do is push the handling of Nothing/Just for all RPC response types down into the `raise` functionality of  the `Convert` module. This response field optionality is an artifact of GRPC/proto which it is sad to have repeated in every rpc-wrapper.